### PR TITLE
GDAL: bump version

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	devel:libz$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:automake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
-	cmd:libtoolize
+	cmd:libtoolize$secondaryArchSuffix
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	autoreconf -fi
+	runConfigure ./configure --with-expat=$expatDir
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+
+	prepareInstalledDevelLibs libgdal
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
-	prepareInstalledDevelLibs libgdal
+	prepareInstalledDevelLib libgdal
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 3]
```diff
@@ -0,0 +1,138 @@
+SUMMARY="The Geospatial Data Abstraction Library"
+DESCRIPTION="GDAL is a translator library for raster and vector geospatial \
+data formats that is released under an X/MIT style OpenSource license by the \
+Open Source Geospatial Foundation. As a library, it presents a single raster \
+abstract data model and vector abstract data model to the calling application \
+for all supported formats. It also comes with a variety of useful \
+commandline utilies for data translation and processing."
+HOMEPAGE="http://www.gdal.org"
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
-HOMEPAGE="http://www.gdal.org"
+HOMEPAGE="http://www.gdal.org/"
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	haiku$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libgeotiff$secondaryArchSuffix
+	lib:libgif$secondaryArchSuffix >= 7.0.0
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
+	lib:libproj$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
 	lib:libgif$secondaryArchSuffix >= 7.0.0
+	lib:libiconv$secondaryArchSuffix
 	lib:libjpeg$secondaryArchSuffix
```
```diff
 	lib:libz$secondaryArchSuffix
-	lib:libiconv$secondaryArchSuffix
 	"
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+	lib:libpng$secondaryArchSuffix
+	lib:libproj$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	gdal${secondaryArchSuffix}_devel = $portVersion compat >= 1.11
+	cmd:gdal_config$secondaryArchSuffix
+	devel:libgdal$secondaryArchSuffix = 20.3.2 compat >= 1.18
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
 PROVIDES_devel="
-	gdal${secondaryArchSuffix}_devel = $portVersion compat >= 1.11
+	gdal${secondaryArchSuffix}_devel = $portVersionCompat
 	cmd:gdal_config$secondaryArchSuffix
-	devel:libgdal$secondaryArchSuffix = 20.3.2 compat >= 1.18
+	devel:libgdal$secondaryArchSuffix = $libVersionCompat
 	"
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+	cmd:automake
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage gdal$secondaryArchSuffix \
+	$libDir/libgdal.so.20.3.2
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
 defineDebugInfoPackage gdal$secondaryArchSuffix \
-	$libDir/libgdal.so.20.3.2
+	$libDir/libgdal.so.$libVersion
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+	cmd:testepsg
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libgeotiff$secondaryArchSuffix
+	lib:libgif$secondaryArchSuffix >= 7.0.0
+	lib:libiconv$secondaryArchSuffix
+	lib:libjpeg$secondaryArchSuffix
+	lib:libpng$secondaryArchSuffix
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
-	lib:libpng$secondaryArchSuffix
+	lib:libpng16$secondaryArchSuffix
```

File: sci-libs/gdal/gdal-2.2.3.recipe [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+	gdal$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libexpat$secondaryArchSuffix
+	devel:libgeotiff$secondaryArchSuffix
+	devel:libgif$secondaryArchSuffix >= 7.0.0
+	devel:libjpeg$secondaryArchSuffix
+	devel:libpng$secondaryArchSuffix
```
Comment (fbrosson) [inline sci-libs/gdal/gdal-2.2.3.recipe]:
```diff
-	devel:libpng$secondaryArchSuffix
+	devel:libpng16$secondaryArchSuffix
```

**Comment by extrowerk** [comment]
It was a big version bump, but i'm not sure if it is compatible with 1.x or not. It would be maybe better to rename the old recipe to gdal1.
Have you got any info about the compatibility, @korli ?

**Comment by waddlesplash** [comment]
OpenCV is the only thing that requires GDAL, so we should be fine.

**Pull request merged**